rosbag2_cpp/test/rosbag2_cpp/test_time_controller_clock.cpp

```diff
@@ -211,6 +211,28 @@ TEST_F(TimeControllerClockTest, interrupted_sleep_returns_false_immediately)
   EXPECT_FALSE(thread_sleep_result);
 }
 
+TEST_F(TimeControllerClockTest, wakeup_sleep_returns_immediately)
+{
+  // Use a bigger value for sleep_time_while_paused
+  auto sleep_time_while_paused = std::chrono::seconds(30);
+  rosbag2_cpp::TimeControllerClock clock(ros_start_time, now_fn, sleep_time_while_paused);
+  clock.pause();
+  auto steady_start = std::chrono::steady_clock::now();
+  std::atomic_bool thread_sleep_result{true};
+  auto sleep_long_thread = std::thread(
+    [&clock, &thread_sleep_result]() {
+      bool sleep_result = clock.sleep_until(clock.now() + RCUTILS_S_TO_NS(10));
+      thread_sleep_result.store(sleep_result);
+    });
+  // Wait enough time for the thread to start sleeping before we try to wake it up
+  std::this_thread::sleep_for(std::chrono::milliseconds(100));
```

Comment:
Ahhh. Unfortunately, this sleep going to cause none-determinism and flakiness in the test.
To make it deterministic will need to add one more API. aka `bool TimeControllerClock::is_in_sleep()` and add one more private variable to detect a state when the timer is in sleep mode.

Reply:
I added `PlayerClock::is_sleeping()`: 190649f9dfc7949b683f96d2ba2dfa113cdd6dec